compiler/packages/babel-plugin-react-compiler/src/Validation/ValidateNoDerivedComputationsInEffects_exp.ts

```diff
@@ -330,12 +361,20 @@ function recordInstructionDerivations(
       case Effect.ConditionallyMutate:
       case Effect.ConditionallyMutateIterator:
       case Effect.Mutate: {
-        if (isMutable(instr, operand)) {
-          context.derivationCache.addDerivationEntry(
-            operand,
-            sources,
-            typeOfValue,
+        if (
+          isMutable(instr, operand) &&
+          context.derivationCache.cache.has(operand.identifier.id)
```

Comment:
If the derivation cache does not already have this entry, we still may want to record a `typeOfValue`. Or is this already handled by your fixpoint iteration logic?

```js
function useHook() {
  const [s, setS] = useState();
  const cb = () => append(arr, s);  // `arr` is mutable here and should get marked as `fromState` 
  const arr = [];
}
```